src/pyhf/pdf.py

```diff
@@ -534,18 +551,34 @@ def constraint_logpdf(self, auxdata, pars):
     def mainlogpdf(self, maindata, pars):
         return self.main_model.logpdf(maindata, pars)
 
-    def logpdf(self, pars, data):
-        try:
-            tensorlib, _ = get_backend()
-            pars, data = tensorlib.astensor(pars), tensorlib.astensor(data)
+    def make_pdf(self, pars):
+        tensorlib, _ = get_backend()
+        pars = tensorlib.astensor(pars)
+
+        bindata = self.nominal_rates.shape[-1]
```

Comment:
I'm guessing `self.nominal_rates` is the shape of `(batch, bins)` right?

Reply:
almost..  but yes bins in the last dimension